src/examples/java/software/amazon/encryption/s3/examples/RangedGetExample.java

```diff
@@ -0,0 +1,131 @@
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.encryption.s3.S3EncryptionClient;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static software.amazon.encryption.s3.utils.S3EncryptionClientTestResources.KMS_KEY_ID;
+import static software.amazon.encryption.s3.utils.S3EncryptionClientTestResources.appendTestSuffix;
+
+public class RangedGetExample {
+
+    // Create a 200 character input string to use as your object in the following examples.
+    private static final String OBJECT_CONTENT = "abcdefghijklmnopqrstABCDEFGHIJKLMNOPQRST" +
+            "abcdefghijklmnopqrstABCDEFGHIJKLMNOPQRST" +
+            "abcdefghijklmnopqrstABCDEFGHIJKLMNOPQRST" +
+            "abcdefghijklmnopqrstABCDEFGHIJKLMNOPQRST" +
+            "abcdefghijklmnopqrstABCDEFGHIJKLMNOPQRST";
+
+    public static void main(final String[] args) {
+        final String bucket = args[0];
+
+        simpleAesGcmV3RangedGet(bucket);
+        aesGcmV3RangedGetOperations(bucket);
+    }
+
+    public static void simpleAesGcmV3RangedGet(String bucket) {
+        final String objectKey = appendTestSuffix("simple-v3-ranged-get-example");
+
+        // Instantiate the S3 Encryption Client by specifying an AES Key with the aesKey builder parameter.
+        // You must also specify the enableLegacyUnauthenticatedModes parameter to enable ranged GET requests.
+        //
+        // This means that the S3 Encryption Client can perform both encrypt and decrypt operations,
+        // and can perform ranged GET requests when a range is provided.
+        S3Client v3Client = S3EncryptionClient.builder()
+                .kmsKeyId(KMS_KEY_ID)
+                .enableLegacyUnauthenticatedModes(true)
+                .build();
+
+        // Call putObject to encrypt the object and upload it to S3
+        v3Client.putObject(PutObjectRequest.builder()
+                                   .bucket(bucket)
+                                   .key(objectKey)
+                                   .build(), RequestBody.fromString(OBJECT_CONTENT));
+
+        // Call getObject to retrieve a range of 10-20 bytes from the object content.
+        ResponseBytes<GetObjectResponse> objectResponse = v3Client.getObjectAsBytes(builder -> builder
+                .bucket(bucket)
+                .range("bytes=10-20")
+                .key(objectKey));
+        String output = objectResponse.asUtf8String();
+
+        // Verify that the decrypted object range matches the original plaintext object at the same range.
+        // Note: The start and end indices of the byte range are included in the returned object.
+        assertEquals(OBJECT_CONTENT.substring(10, 20 + 1), output);
+
+        // Cleanup
+        v3Client.deleteObject(builder -> builder.bucket(bucket).key(objectKey));
+        v3Client.close();
+    }
+
+    public static void aesGcmV3RangedGetOperations(String bucket) {
```

Comment:
nit: let's add a comment mentioning that this example demonstrates how various unusual ranged get cases are handled